jobs/release-microk8s/release-microk8s.groovy

```diff
@@ -0,0 +1,131 @@
+@Library('juju-pipeline@master') _
+
+
+pipeline {
+    agent {
+        label "runner-${params.ARCH}"
+    }
+    /* XXX: Global $PATH setting doesn't translate properly in pipelines
+     https://stackoverflow.com/questions/43987005/jenkins-does-not-recognize-command-sh
+     */
+    environment {
+        PATH                 = "${utils.cipaths}"
+        ARCH                 = "${params.ARCH}"
+        JUJU_CLOUD           = "aws/us-east-1"
+        K8STEAMCI            = credentials('k8s_team_ci_lp')
+        CDKBOT_GH            = credentials('cdkbot_github')
+        LPCREDS              = credentials('launchpad_creds')
+        CHARM_CREDS          = credentials('charm_creds')
+        JUJU_CREDS           = credentials('juju_creds')
+        JUJU_CLOUDS          = credentials('juju_clouds')
+        SSOCREDS             = credentials('sso_token')
+        SNAPCRAFTCREDS       = credentials('snapcraft_creds')
+        SNAPCRAFTCPCCREDS    = credentials('snapcraft_cpc_creds')
+        AWS_CREDS            = credentials('aws_creds')
+        SURL_CREDS           = credentials('surl-creds')
+        AWSIAMARN            = credentials('aws-iam-arn')
+        CDKBOTSSHCREDS       = credentials('cdkbot_ssh_rsa')
+        K8STEAMCI_GPG_PUB    = credentials('deb-gpg-public')
+        K8STEAMCI_GPG_PRIVATE= credentials('deb-gpg-private')
+        K8STEAMCI_GPG_KEY    = credentials('deb-gpg-key')
+
+    }
+    options {
+        ansiColor('xterm')
+        timestamps()
+    }
+    stages {
+        stage("Setup Tracks") {
+            steps {
+                script {
+                    if (params.ARCH_TRACK == "all") {
+                        all_tracks = ["beta", "stable", "pre-release"]
+                    } else {
+                        all_tracks = [params.ARCH_TRACK.tokenize('/').last()]
+                    }
+                }
+            }
+        }
+        stage("Snapcraft login") {
+            steps {
+                sh "snapcraft login --with ${SNAPCRAFTCREDS}"
+            }
+        }
+        stage("Setup tox environment") {
+            steps {
+                sh """
+                tox -e py38 -- python -c 'print("Tox Environment Ready")'
+                """
```

Comment:
this setups up the tox environment once, so every track run doesn't need it re-invoked